src/backend_model_instance.cc

```diff
@@ -358,35 +371,32 @@ TritonModelInstance::CreateInstance(
     RETURN_IF_ERROR(model->Server()->GetRateLimiter()->RegisterModelInstance(
         local_instance.get(), rate_limiter_config));
     RETURN_IF_ERROR(local_instance->SetBackendThread(
-        kind, device_id, device_blocking, device_to_thread_map));
+        kind, device_id, model->DeviceBlocking()));
   }
 
-  RETURN_IF_ERROR(model->AddInstance(std::move(local_instance), passive));
```

Comment:
I think it would be better to decouple the creation and registration of the instances.

Reply:
Update commit: [Decouple registration from instance creation](https://github.com/triton-inference-server/core/pull/189/commits/0565adcaf6e6b7b7600d60777189468a5eb07c5e)